nautobot/utilities/views.py

```diff
@@ -151,3 +187,1051 @@ def get_return_url(self, request, obj=None):
 
         # If all else fails, return home. Ideally this should never happen.
         return reverse("home")
+
+
+class NautobotViewSetMixin:
+
+    routes = []
+    model = None
+    queryset = None
+    table = None
+    template_name = None
+    filterset = None
+    filterset_form = None
+    import_form = None
+    prefetch_related = []
+
+    def detail_queryset(self):
+        return self.model.objects.all()
+
+    def table_queryset(self):
+        return self.model.objects.prefetch_related(*self.prefetch_related)
+
+    def define_routes(self):
+        return []
+
+    def get_extra_context(self, request, view_type, instance=None):
+        """
+        Return any additional context data for the template.
+        request: The current request
+        instance: The object being viewed
+        """
+        return {}
+
+    def get_template_name(self, view_type):
+        # Use "<app>/<model>_<view_type> if available, else fall back to generic templates
+        model_opts = self.model._meta
+        app_label = model_opts.app_label
+        if view_type == "detail":
+            return f"{app_label}/{model_opts.model_name}.html"
```

Comment:
Per: https://github.com/nautobot/nautobot/pull/1902/files#r914237010

Let's remove this. We should require `{app_label}/{model_name}_{view_type}.html` for all view types to adopt `NautobotViewSet`.